posthog/queries/funnels/funnel_event_query.py

```diff
@@ -102,15 +102,16 @@ def get_query(
         sample_clause = "SAMPLE %(sampling_factor)s" if self._filter.sampling_factor else ""
         self.params.update({"sampling_factor": self._filter.sampling_factor})
 
+        all_fields = f"{', '.join(_fields)} {{extra_select_fields}}"
+
         # KLUDGE: Ideally we wouldn't mix string variables with f-string interpolation
         # but due to ordering requirements in functions building this query we do
         # things like this for now but should do a larger refactor to get rid of it
         query = f"""
-            SELECT {', '.join(_fields)}
-            {{extra_select_fields}}
+            SELECT {all_fields}
             FROM events {self.EVENT_TABLE_ALIAS}
             {sample_clause}
-            {self._get_person_ids_query()}
+            {self._get_person_ids_query(relevant_events_conditions=entity_query + date_query)}
```

Comment:
Using `prop_query` would be yet another major reduction in memory usage in queries with narrowly defined filters, but I tried it and this is currently pretty painful because suddenly this deep filtering query needs not just conditions, but also fields. I found this to be tricky to get right. Likely will be easier to reason about that when this optimization is part of a HogQL visitor, rather than this query concat monster.